src/core/currency/wallet/currency-wallet-cleaners.ts

```diff
@@ -193,6 +207,47 @@ export function asIntegerString(raw: unknown): string {
 // file cleaners
 // ---------------------------------------------------------------------
 
+export const asNumeralString = (raw: any): string => {
+  if (typeof raw !== 'string') throw new Error('Not type string')
+  if (isNaN(Number(raw))) throw new Error('Does not parse as a number')
+  return raw
+}
```

Comment:
The intent was decimal support although unneeded for its current use. Once Sam merges his biggystring PR we'll have full scientific notation support. For now I'll change to `asIntegerString`

Reply:
Even if biggystring supports full exponent syntax, I think it would still make sense to require integers here - these are native amounts, so if a decimal point sneaks in, we have a bug.